pkgs/development/python-modules/lsp-tree-sitter/default.nix

```diff
@@ -14,21 +15,24 @@
 
 buildPythonPackage rec {
   pname = "lsp-tree-sitter";
-  version = "0.0.15";
+  version = "0.0.16";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "neomutt";
     repo = "lsp-tree-sitter";
-    rev = version;
-    hash = "sha256-yzScgix3BtSCBzlDoE1kMYGtVzkup/+ZK9L1C7VA3do=";
+    rev = "refs/tags/${version}";
+    hash = "sha256-nRzyVZFgb08M0I+xmhuX1LDwPsghlBLdZ2Ou8stKzk0=";
   };
 
   build-system = [
     setuptools-generate
     setuptools-scm
+    pythonRelaxDepsHook
   ];
 
+  pythonRelaxDeps = lib.optionals (lib.versionOlder tree-sitter.version "0.22") [ "tree-sitter" ];
```

Comment:
Can we make it unconditional?

Reply:
> Can we make it unconditional?

Hmm I'm not sure about this... I think it is good for us to know about such constraints that upstream sets. So for example in a future attempt to update `lsp-tree-sitter`, if `tree-sitter` will have to be updated too, it will tell us so. Also, the fact that `autotools-language-server` works with `lsp-tree-sitter` 0.0.16 is surprising TBH. I was even thinking of changing the `python311`'s `packageOverrides` function in `autotools-language-server` to include this relaxation, and not add it here.

Reply:
> > Can we make it unconditional?
> 
> Hmm I'm not sure about this... I think it is good for us to know about such constraints that upstream sets. So for example in a future attempt to update `lsp-tree-sitter`, if `tree-sitter` will have to be updated too, it will tell us so. Also, the fact that `autotools-language-server` works with `lsp-tree-sitter` 0.0.16 is surprising TBH. I was even thinking of changing the `python311`'s `packageOverrides` function in `autotools-language-server` to include this relaxation, and not add it here.

I removed that `pythonRelaxDeps` from here, and added it in the `packageOverrides` of `autotools-language-server`.